lib/ansible/modules/network/asa/asa_facts.py

```diff
@@ -0,0 +1,361 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = """
+---
+module: asa_facts
+version_added: "2.4.0"
+author: "Narate Pokasub (@npokasub)"
+short_description: Collect facts from remote devices running Cisco ASA
+description:
+  - Collects a base set of device facts from a remote device that
+    is running Cisco ASA.  This module prepends all of the
+    base network fact keys with C(ansible_net_<fact>).  The facts
+    module will always collect a base set of facts from the device
+    and can enable or disable collection of additional facts.
+extends_documentation_fragment: asa
+notes:
+  - Tested against ASAv 9.8.1
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset.  Possible values for this argument include
+        all, hardware, config, and interfaces.  Can specify a list of
+        values to include a larger subset.  Values can also be used
+        with an initial C(M(!)) to specify that a specific subset should
+        not be collected.
+    required: false
+    default: '!config'
+"""
+
+EXAMPLES = """
+# Collect all facts from the device
```

Comment:
Please use `- name: Collect all facts from the device` rather than a comment.

Same on the rest of the examples